assignments/array-methods.js

```diff
@@ -53,31 +53,77 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 {"id":49,"first_name":"Bel","last_name":"Alway","email":"balway1c@ow.ly","shirt_size":"S","company_name":"Voolia","donation":107},
 {"id":50,"first_name":"Shell","last_name":"Baine","email":"sbaine1d@intel.com","shirt_size":"M","company_name":"Gabtype","donation":171}];
 
+
+//runners
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+    runners.forEach(function (runner) {
+        return fullName.push(runner.first_name + " " + runner.last_name);
+    });
+
+
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
+    runners.map(function (runner) {
+        return allCaps.push(runner.first_name.toUpperCase());
+    });
+
+
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+    runners.filter(function (runner) {
+        if (runner.shirt_size === 'L') {
+          largeShirts.push(runner);
+        }
+    });
+
+
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+    ticketPriceTotal = runners.reduce(function (total, runner) {
+        return total + runner.donation;
+    }, 0); // <-- that little zero makes it tick.
+
+    
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1 Well the power hungry director forgot to tell the city about the 5k. So none of the route has been cleared for use. He needs to email all the runners and let them know that the 5k has been moved to a farm outside of town.
+
+let emailList = [];
+    runners.forEach(function (runner) {
+        return emailList.push(runner.email);
+    });
+
+console.log(emailList)
+
+// Problem 2 No one is sure why this person is in charge any more, but it's too late now. He needs a list of people that donated less than 40 units of generic currency. So he can bully them.
+let lessMoney = [];
+    runners.filter(function (runner) {
```

Comment:
great job, for a stretch you could also turn the runner.donation into a parsed integer so that you can compare number types. This happens a lot in programming where you'll get one data type but need another so you'll have to convert it.